src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/ReindexJobCompletedHandler.cs

```diff
@@ -0,0 +1,77 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using MediatR;
+using Microsoft.Health.Fhir.Core.Exceptions;
+using Microsoft.Health.Fhir.Core.Features.Definition;
+using Microsoft.Health.Fhir.Core.Features.Search;
+using Microsoft.Health.Fhir.Core.Features.Search.Registry;
+using Microsoft.Health.Fhir.Core.Features.Security;
+using Microsoft.Health.Fhir.Core.Features.Security.Authorization;
+using Microsoft.Health.Fhir.Core.Messages.Reindex;
+
+namespace Microsoft.Health.Fhir.Core.Features.Operations.Reindex
+{
+    public class ReindexJobCompletedHandler : IRequestHandler<ReindexJobCompletedRequest, ReindexJobCompletedResponse>
+    {
+        private readonly ISupportedSearchParameterDefinitionManager _searchParameterDefinitionManager;
+        private readonly ISearchParameterRegistry _searchParameterRegistry;
+        private readonly IFhirAuthorizationService _authorizationService;
+
+        public ReindexJobCompletedHandler(
+            ISupportedSearchParameterDefinitionManager searchParameterDefinitionManager,
+            ISearchParameterRegistry searchParameterRegistry,
+            IFhirAuthorizationService fhirAuthorizationService)
+        {
+            EnsureArg.IsNotNull(searchParameterDefinitionManager, nameof(searchParameterDefinitionManager));
+            EnsureArg.IsNotNull(searchParameterRegistry, nameof(searchParameterRegistry));
+            EnsureArg.IsNotNull(fhirAuthorizationService, nameof(fhirAuthorizationService));
+
+            _searchParameterDefinitionManager = searchParameterDefinitionManager;
+            _searchParameterRegistry = searchParameterRegistry;
+            _authorizationService = fhirAuthorizationService;
+        }
+
+        public async Task<ReindexJobCompletedResponse> Handle(ReindexJobCompletedRequest message, CancellationToken cancellationToken)
+        {
+            if (await _authorizationService.CheckAccess(DataActions.Reindex) != DataActions.Reindex)
```

Comment:
Just curious how the authorization flow will look like here since this request is being generated internally

Reply:
You are correct, this is a problem.  I added in the security check because we have a unit test which runs against all classes which implement IRequestHandler and it failed without the check.  However, this authorization does not work on internal call.  I had turned off security in appsettings, so I missed it.  I decided it makes more sense to change the call to a simple method call, rather than a mediatr message.  Which avoids the problem.